exchanges/bitstamp/bitstamp_wrapper.go

```diff
@@ -762,6 +769,8 @@ func (b *Bitstamp) WithdrawFiatFunds(ctx context.Context, withdrawRequest *withd
 // WithdrawFiatFundsToInternationalBank returns a withdrawal ID when a
 // withdrawal is submitted
 func (b *Bitstamp) WithdrawFiatFundsToInternationalBank(ctx context.Context, withdrawRequest *withdraw.Request) (*withdraw.ExchangeResponse, error) {
+	withdrawRequest.Exchange = b.Name
+	withdrawRequest.Type = withdraw.Fiat
```

Comment:
I don't usually like mutating the request values, these should be set prior to this function being called. Could you please update withdrawRequest.Validate function to check for these values and return an error, and remove these assignments?

Reply:
FWIW: I don't see the benefit of pushing the responsibility down to the consumer to set values that are functionally immutable.
In any exchange wrapper, making a caller set .Exchange seems wrong.
And making them set a request type that's implied by the function seems wrong too.
Feel free to ignore my opinion on this though.
I suspect that if I pull on this thread too hard I'm going to start questioning withdrawRequest in general.

Reply:
I have left this one for you guys to decide on the best solution

Reply:
I would prefer to not override the user's request. It would be better to tell them why they're wrong via `Validate` or another error than for them to send an incorrect request and have it work which then goes against their intention (even though it was wrong). Since its dealing with real money, we shouldn't override things like this. 

The exchange name I don't mind as much

Reply:
Okay. Agreed.
The issue here is about the type, and the fact it's in a shared request type.
Since this was a drive-by, let's back it out, and have the tests calling this set `.Type` and `.Exchange`
It should all be fixed in one go. There's undoubtedly other issues that a fresh birdseye view would fix.

Reply:
I have removed the request values from the wrapper and added them to the test instead